.github/workflows/cron-daily.yml

```diff
@@ -0,0 +1,74 @@
+# Runs every day to check if Freenet's fred repository has changed.
+# If yes it runs the CI tests to see if the changes at fred broke anything at Freetalk.
+# If no it does nothing.
+# NOTICE: This only runs on xor-freenet's repository currently, see the comments below.
+name: CI cronjobs, daily
+
+# TODO: Code quality: Currently Freetalk only has branch "master" so the cronjobs only run there.
+# Once there's also a development branch run the cronjobs on both branches.
+
+on:
+  schedule:
+    - cron: 07 19 * * * # At 19:07 every day
```

Comment:
This is pretty dangerous, because it runs a lot of tests. Please reduce it to at most one run per week, on only a single Java version (I’m not sure whether it does that).

Just replace the final * with 1 to only run on a single day of the week.

Reply:
Thanks for the feedback!
:)

> This is pretty dangerous, because it runs a lot of tests.

Oh it seems you misread what this does, here's what actually happens:

- It **only does something on my personal GitHub account**. On all other accounts it will get cancelled right away due to the two `if:`. Read the full script, the `if:` aren't in the cron-part but rather below that.  
You can see this on your repo, the runs are marked with a grey icon whose tooltip says "This workflow run was skipped", and the two existing ones only took 2 secs and 3 secs:
https://github.com/ArneBab/plugin-Freetalk-staging/actions/workflows/cron-daily.yml

- **Even on my account it does NOT run the tests strictly every day**. What it does every day is checking if fred branch next has changed and only running the tests then. Since fred only receives commits in O(one or more weeks) it will effectively only run the tests every few weeks.  So on my repo it only takes ~15 seconds of execution time per day normally to determine that fred didn't change. On other repos the execution time is even lower because:

- The check if fred has changed also **only runs on my account**. 

So effectively for all repos but mine it will consume 2-3 seconds per day, which is approximately zero as compared to other projects which constantly push commits and trigger test runs :)

 > Releasing currently takes at least two weeks from merging to shipping, so testing once per week should be sufficient to catch any kind of breakage before the release goes live.
 
My total personal obligations are 50-70 hours per week so I need to notice unexpected urgent issues ASAP in order to be able to allocate time to them.

Since the failures will notify me, not the fred developers (see the comments in the script for why that was chosen), and I will also have to write up something to the fred developers which would justify them having to repair their code, I need to be alerted early enough to deal with this.

Weekly runs would mean in the worst case I only have 1 week for that, which would put high pressure on me.

So given that GitHub would likely complain to **me** in the end if the runtime is too high as it runs the tests on my account I would please like to keep this as-is.